### PR TITLE
[dompurify] Add typing for factory function

### PR DESCRIPTION
**Pull request by hgenru**
@See https://github.com/cure53/DOMPurify/issues/323

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

### CODE REVIEW

File: types/dompurify/index.d.ts [hunk 1]
```diff
@@ -3,26 +3,37 @@
 // Definitions by: Dave Taylor <http://davetayls.me>, Samira Bazuzi <https://github.com/bazuzi>, FlowCrypt <https://github.com/FlowCrypt>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-export as namespace DOMPurify;
+export = createDOMPurify
```
Comment (amcasey) [inline types/dompurify/index.d.ts]: It looks like the new form isn't consumable as a global.  Was that intentional?

Comment (hgenru) [reply types/dompurify/index.d.ts]: @amcasey can you explain the difference? I am not familiar with typings, I just write as I know and test it.

Comment (amcasey) [reply types/dompurify/index.d.ts]: I think of the `as namespace` version as being a way to make the declarations available in non-modules (e.g. webpages).  There's a more thorough explanation here: https://www.typescriptlang.org/docs/handbook/modules.html#umd-modules

Comment (hgenru) [reply types/dompurify/index.d.ts]: Should I rename DOMPurify interface to DOMPurifyI and export createDOMPurify as DOMPurify namespace?

Comment (hgenru) [reply types/dompurify/index.d.ts]: I figured out. What you think, maybe we should add this into the FAQ?

**Comment by quentez** [comment]
@hgenru Hi. Since updating this typings, we're not able to import the other interfaces that it previously contained anymore.

This worked before:

```
import DOMPurify, {Config, HookEvent, HookName} from 'dompurify';
```

Could this be because of this change?

**Comment by hgenru** [comment]
@quentez, very likely. Config, HookEvent, HookName wasn't exported clearly and I have no idea why this worked before.

Sorry, I had no plans to break your code. Can you add a fix and new tests? It's will be very nice.

**Comment by crimx** [comment]
Also break my code. Cannot import the namespace anymore. These types are useful when writing wrappers for dompurify.

**Comment by crimx** [comment]
Changing to  `export declare interface DOMPurifyI {` seems to work. Can you verify that? @hgenru

**Pull request merged**